Cython/Compiler/ExprNodes.py

```diff
@@ -8267,7 +8268,9 @@ def generate_sequence_packing_code(self, code, target=None, plain=False):
     def generate_subexpr_disposal_code(self, code):
         if self.mult_factor and self.mult_factor.type.is_int:
             super().generate_subexpr_disposal_code(code)
-        elif self.type is tuple_type and (self.is_literal or self.slow):
+        elif ((self.type is tuple_type or self.type is list_type) and
+                (self.is_literal or self.slow) and
+                len(self.args) > 0):
```

Comment:
You left out the multiply part that appears in the other condition. I'd rather have the whole condition extracted and reused than having it (partly) duplicated. It's really complex and (evidently) easy to forget about.

Reply:
The `plain` argument to `generate_sequence_packing_code` makes it quite hard to reason about what we need to do in `generate_subexpr_disposal_code` so I just recorded what we actually did and based it on that.